src/index.ts

```diff
@@ -1,923 +1,1064 @@
+          }
+        } catch (err) {
+          if (err.message.toLowerCase().includes("already in interface"))
+            log.warn("Cannot Activate App: Installed App ID (%s) is already added to hosted interface", installed_app_id);
+          else {
+            log.error("Failed during 'activateApp': %s", String(err));
+            throw err;
+          }
+        }
+
+        // Attach App to Interface - Connect app to hosted interface and start app (ie: spin up all cells within app bundle)
+        try {
+          let hosted_port
+
+          if ((this.opts.hosted_port_number === 0 || this.opts.hosted_port_number) && this.opts.mode === Envoy.DEVELOP_MODE) {
+            log.info("Defaulting to port provided in opts config.  Attaching App to port (%s)", this.opts.hosted_port_number);
+            // NOTICE: MAKE SURE THIS PORT IS SET TO THE WS PORT EXPECTED IN THE UI
+            hosted_port = this.opts.hosted_port_number;
+          } else {
+            hosted_port = this.conductor_opts.interfaces.hosted_port;
+          }
+
+          log.info("Starting installed-app (%s) on port (%s)", installed_app_id, hosted_port);
+
+          adminResponse = await this.callConductor("master", 'attachAppInterface', { port: hosted_port });
+
+          if (adminResponse.type !== "success") {
+            log.error("Conductor 'attachAppInterface' returned non-success response: %s", adminResponse);
+            failed = true
+            throw (new HoloError(`Failed to complete 'attachAppInterface' for installed_app_id'${installed_app_id}'.`)).toJSON();
+          }
+        } catch (err) {
+          if (err.message.toLowerCase().includes("already active"))
+            log.warn("Cannot Start App: Intalled-app (%s) is already started", installed_app_id);
+          else {
+            log.error("Failed during 'attachAppInterface': %s", String(err));
+            throw err;
+          }
+        }
+
+      } catch (err) {
+        failed = true;
+        log.error("Failed during DNA processing for Agent (%s) HHA ID (%s): %s", agent_id, hha_hash, String(err));
+      }
+
+      if (failed === true) {
+        // TODO: Rollback cells that were already created << check to see if is already being done in RSM.
+        log.error("Failed during sign-up process for Agent (%s) HHA ID (%s): %s", agent_id, hha_hash, failure_response);
+
+        return failure_response;
+      }
+
+      // - return success
+      log.normal("Completed sign-up process for Agent (%s) HHA ID (%s)", agent_id, hha_hash);
+      // TODO: Update to match hhdt success message in both envoy and chaperone
+      // return  new Package( true, { "type": "success" });
+      return true;
+    }, this.opts.NS);
+
+
+    // Chaperone AppInfo Call to Envoy Server
+    this.ws_server.register("holo/app_info", async ({ installed_app_id }) => {
+      this.request_counter++;
```

Comment:
Why do we have a request_counter?

Reply:
This line incrementing the counter is residue from when I was using the counter for the request_id.  I'll remove it.  

The counter, however, already existed in the Envoy class, but does not seem to be used anywhere else.  So I cannot answer the question as to why it was created in the first place.  Happy to remove that as well if you agree.

Reply:
Yes, If it's not used, I agree